File: solutions/patterns/star-patterns/13-hollow-diamond/solution.cpp
#include <iostream>
using namespace std;

int main()
{
    int n;

    // Input odd number only
    do
    {
        cout << "Enter an odd N value: ";
        cin >> n;
        if (n % 2 == 0)
        {
            cout << "Please enter an odd number!\n";
        }
    } while (n % 2 == 0);

    cout << "\nPrinting the inverted hollow pyramid...\n\n";

    for (int row = 1; row <= n; row++)
    {
        int middleOfDiamond = (n / 2) + 1;

        if (row <= middleOfDiamond)
        {
            // TOP HALF (already correct)
            for (int col = 1; col <= (middleOfDiamond - row); col++)
            {
                cout << " ";
            }

            cout << "*";

            if (row > 1)
            {
                for (int space = 1; space <= (2 * row - 3); space++)
                {
                    cout << " ";
                }
                cout << "*";
            }

            cout << endl;
        }
        else
        {
            // ✅ FIXED BOTTOM HALF (MIRROR OF THE TOP)
            int mirrorRow = n - row + 1;

            for (int col = 1; col <= (middleOfDiamond - mirrorRow); col++)
            {
                cout << " ";
            }

            cout << "*";

            if (mirrorRow > 1)
            {
                for (int space = 1; space <= (2 * mirrorRow - 3); space++)
                {
                    cout << " ";
                }
                cout << "*";
            }

            cout << endl;
        }
    }

    return 0;
}

File: solutions/patterns/star-patterns/13-hollow-diamond/solution.java
import java.util.*;

class solution {

    public static void main(String args[]) {
        int n = 4;
        int s = 1;
        for (int i = 1; i <= 2 * n - 1; i++) {
            // top pyramid
            if (i <= n) {

                for (int j = 1; j <= n - i; j++) {
                    System.out.print(" ");
                }
                for (int j = 1; j <= i; j++) {
                    if (j == 1 || j == i) {
                        System.out.print("* ");
                    } else {
                        System.out.print("  ");
                    }
                }

            } else {
                // bottom pyramid
                for (int j = 1; j <= s; j++) {
                    System.out.print(" ");

                }
                s++;
                for (int j = 1; j <= 2 * n - i; j++) {
                    if (j == 1 || j == 2 * n - i) {
                        System.out.print("* ");
                    } else {
                        System.out.print("  ");
                    }

                }
            }

            System.out.println();
        }
    }
}